### DISCUSSION TITLE
DataTable loading slow on 200 rows

### DISCUSSION THREAD

**Discussion by mdimitrov94** [primefaces/community#1624] [PrimeReact] [answered]
Hello.
I wanted to reach out regarding an issue I've encountered with the DataTable component in PrimeReact. Specifically, I've noticed significant performance degradation when using lazy loading in tables with a large number of rows.

The problem arises when the user selects more than 200 rows in lazy loading mode. At this point, the page starts to lag noticeably, and the user experience becomes less than ideal. Furthermore, when dealing with around 2000 rows, the page load time exceeds 5 seconds, which is unacceptable for most user interactions.

**Comment by melloware** [comment]
200 rows visible without paging?

**Reply by mdimitrov94** [reply]
When implementing paging, the issue arises when attempting to increase the number of rows beyond 200. With 2000 rows, the loading time extends to 5-6 seconds, especially when sorting is applied. However, when the table component is commented out, the request completes in less than 1 second, indicating that the bottleneck is not on the backend.

**Reply by melloware** [reply]
Yes so you are loading all that data into the HTML page.  So paging only displays 10-25 rows etc. But when you show 200 it is re-rendering 200 rows.  The bottleneck is definitely in the UI.    See these open issues: https://github.com/primefaces/primereact/labels/Type%3A%20Performance

**Reply by mdimitrov94** [reply]
so you are saying it's not certain when this will be fixed or will it be fixed at all ?

**Reply by melloware** [reply]
@mdimitrov94 i don't work for PrimeTek just an open source contributor.  Ticket is open so hopefully it will be fixed but no timeline. If you need immediate support I suggest PrimeReact PRO support.

**Reply by acc-cassio** [reply, marked as answer]
@mdimitrov94 I have solved the issue with the selection. The PR is waiting for review, see https://github.com/primefaces/primereact/pull/7425. In the meanwhile, you can clone my repo and use it in your project: https://github.com/acc-cassio/custom-primereact/
